### ISSUE TITLE
N803 Argument name `_` should be lowercase - false positive?

### ISSUE THREAD

**Issue by LefterisJP** [charliermarsh/ruff#619]
I think that N803 is a false positive here. 
```python
        activate_premium=lambda _: None,
```

It's a common patttern to use `_` when there should be no argument name.
